esp/services/ws_fs/ws_fsBinding.cpp

```diff
@@ -393,12 +393,16 @@ int CFileSpraySoapBindingEx::downloadFile(IEspContext &context, CHttpRequest* re
         if (!context.validateFeatureAccess(FILE_SPRAY_URL, SecAccess_Full, false))
             throw MakeStringException(ECLWATCH_FILE_SPRAY_ACCESS_DENIED, "Failed to download file. Permission denied.");
 
-        StringBuffer netAddressStr, osStr, pathStr, nameStr;
+        StringBuffer netAddressStr, osStr, pathStr, nameStr, dropZoneName;
         request->getParameter("NetAddress", netAddressStr);
         request->getParameter("OS", osStr);
         request->getParameter("Path", pathStr);
         request->getParameter("Name", nameStr);
+        request->getParameter("DropZoneName", dropZoneName);
 
+        const char *dropZoneId = isContainerized() ? dropZoneName.str() : netAddressStr.str();
```

Comment:
on reflection, the aim is to support planes in bare-metal too, so I think it would be better to not make this conditional on isContainerized().

And, in bare-metal a named dropzone will have been automatically created from the environment.

I think this and similar code should use the dropZoneName in the request, and if a legacy IP/hostname is supplied, convert it (look it up) to it's drop zone name.
There is similar code that does this for despray here : https://github.com/hpcc-systems/HPCC-Platform/blob/2de4ad0b7f8fec292513bf067b9edb35e400e306/esp/services/ws_fs/ws_fsService.cpp#L2499

We should move that code to a helper function (dautils) e.g. to findHostedPlane(const char *host) 
Then code here and in other similar places can look at the supplied dropzonename and/or host/netaddr and lookup the dropzone name.

Then all the calls to getDropZoneScopePerm* can unify into only accepting a dropzone name.

Reply:
Do we also need a findDropZoneName(const char *host) for non-plane dropzones?

Reply:
For a non-plane dropzone, is it possible to allow a user to access one dropzone server, but, not access another dropzone server?

Reply:
they could.. if we've constructed a external scope with host/ip - they could LDAP scope those if they wanted to.